src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -288,6 +288,27 @@ fn check_contract_of(checker: &mut TypeChecker, args: &[SymbolicExpression], con
     Ok(TypeSignature::PrincipalType)
 }
 
+fn check_principal_of(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(1, args)?;
+    checker.type_check_expects(&args[0], context, &BUFF_33)?;
+    Ok(TypeSignature::PrincipalType)
+}
+
+fn check_secp256k1_recover(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(2, args)?;
+    checker.type_check_expects(&args[0], context, &BUFF_32)?;
+    checker.type_check_expects(&args[1], context, &BUFF_65)?;
+    Ok(BUFF_33)
+}
+
+fn check_secp256k1_verify(checker: &mut TypeChecker, args: &[SymbolicExpression], context: &TypingContext) -> TypeResult {
+    check_argument_count(3, args)?;
+    checker.type_check_expects(&args[0], context, &BUFF_32)?;
+    checker.type_check_expects(&args[1], context, &BUFF_65)?;
+    checker.type_check_expects(&args[2], context, &BUFF_33)?;
+    Ok(TypeSignature::BoolType)
```

Comment:
Along the same vein, should we wrapped this `BoolType` in an Optional / Response, so developers have a way to handle a signature failing at verifying, and a bad construct.

Reply:
hmm, I thought I had commented on this already.  I can't figure out a use case where a contract could recover from a signature not verifying for whatever reason.  The only thing that matters is whether the function returns true or not.  Based on some other feedback, it will return false more than before, allowing the contract to maybe try an other signature and ignore that one...

Reply:
Returning true/false works for me -- the only reason we may want to use a response type is if we want err codes for the signature failures.